components/datatypes/consent-preferences.schema.json

```diff
@@ -0,0 +1,261 @@
+      "description": "Through which interface did the user provide this preference value",
+      "type": "string",
+      "maxLength": 15
+    },
+    "consent-fields": {
+      "type": "object",
+      "properties": {
+        "xdm:v": {
+          "description": "The value specified by the user as their consent choice or the basis of processing that precludes the need for user consent",
+          "$ref": "#/definitions/choice-value"
+        }
+      },
+      "required": ["xdm:v"]
+    },
+    "personalization-fields": {
+      "type": "object",
+      "properties": {
+        "xdm:v": {
+          "$ref": "#/definitions/choice-value"
+        }
+      },
+      "required": ["xdm:v"]
+    },
+    "marketing-fields": {
+      "type": "object",
+      "properties": {
+        "xdm:v": {
+          "$ref": "#/definitions/choice-value"
+        },
+        "xdm:reason": {
+          "title": "Opt out reason",
+          "type": "string",
+          "description": "Reason for opting out",
+          "maxLength": 16
+        }
+      },
+      "required": ["xdm:v"]
+    },
+    "choices": {
+      "properties": {
+        "xdm:collect": {
+          "title": "Data Collection",
+          "description": "Data collection is permitted",
+          "$ref": "#/definitions/consent-fields"
+        },
+        "xdm:share": {
+          "title": "Share Data",
+          "description": "Sharing of user's Data with 2nd or 3rd parties is permitted",
+          "$ref": "#/definitions/consent-fields"
+        },
+        "xdm:adID": {
+          "title": "Use Advertiser ID",
+          "description": "The Advertiser ID (IDFA / AAID) can be used to link user across apps on this device",
+          "$ref": "#/definitions/consent-fields"
+        },
+        "xdm:personalize": {
+          "title": "Personalization Preferences",
+          "description": "User's Personalization Preferences",
+          "type": "object",
+          "properties": {
+            "xdm:any": {
+              "title": "General Personalization Preference",
+              "description": "Is permission granted to perform any type of personalization",
+              "$ref": "#/definitions/personalization-fields"
+            },
+            "xdm:content": {
+              "title": "Personalize Content",
+              "description": "Allow personalization of the content on my site or in my apps",
+              "$ref": "#/definitions/personalization-fields"
+            }
+          }
+        },
+        "xdm:marketing": {
+          "title": "Marketing Preferences",
+          "description": "User's Direct Marketing Preferences",
+          "type": "object",
+          "properties": {
+            "xdm:preferred": {
+              "title": "Preferred Channel",
+              "description": "User's preferred channel for receiving communications",
+              "type": "string",
+              "enum": [
+                "email",
+                "push",
+                "in_app",
```

Comment:
As I look through XDM is seems like we have some enums camelCase, some TitleCase, and some snake_case. Is there an XDM standard outlined somewhere?

Reply:
I asked about this previously and was told all are in common use with no standard. I suppose I could switch these to camelCase to eliminate the underscores, reducing the JSON by a byte each time one occurs.

Reply:
I don't personally have much of a preference. I was just hoping it was standardized. Thanks for checking into this.